### ISSUE TITLE
GRIB or BUFR files are recognised as unknown type

### ISSUE THREAD

**Issue by sandorkertesz** [ecmwf/earthkit-data#123]
### What happened?

It happens when a GRIB or BUFR file does not start with the "GRIB" or "BUFR" bytes, but they contain some padding or data transmission headers. Then `from_source` assigns the `unknown` type to them.

### What are the steps to reproduce the bug?

Use 

```
 wget https://github.com/ecmwf/pdbufr/raw/master/tests/sample-data/M02-HIRS-HIRxxx1B-NA-1.0-20181122114854.000000000Z-20181122132602-1304602.bufr --output-document=hirs.bufr
```
then

```python
import earthkit.data
ds  = earthkit.data.from_source("file", "hirs.bufr")
```

### Version

any

### Platform (OS and architecture)

any

**Comment by juntyr** [comment]
I just stumbled across this issue. I was trying to load the GRIB file from https://apps.ecmwf.int/ifs-experiments/rd/guxf/ using `data = earthkit.data.from_source("file", "data.grib")`, but the data is not recognised as a GRIB file. Would it be possible to add some option to force the selection of a reader / make it just based on the file extension?

**Comment by sandorkertesz** [comment]
@juntyr, this was fixed in #126 (released as part of 0.2.0). It added the `reader-type-check-bytes` settings with a default value of 64. I checked your data and earthkit-data can read it with the default settings! I wonder what eartkit-data version you are using and how you configured `reader-type-check-bytes`. 

<img width="579" alt="image" src="https://github.com/user-attachments/assets/2b1eaf7d-f3c1-467b-8ab1-82dc8a0a6a03">

**Comment by juntyr** [comment]
> @juntyr, this was fixed in #126 (released as part of 0.2.0). It added the `reader-type-check-bytes` settings with a default value of 64. I checked your data and earthkit-data can read it with the default settings! I wonder what eartkit-data version you are using and how you configured `reader-type-check-bytes`.
> <img alt="image" width="579" src="https://private-user-images.githubusercontent.com/9033020/374248587-2b1eaf7d-f3c1-467b-8ab1-82dc8a0a6a03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzODc0NTEsIm5iZiI6MTcyODM4NzE1MSwicGF0aCI6Ii85MDMzMDIwLzM3NDI0ODU4Ny0yYjFlYWY3ZC1mM2MxLTQ2N2ItOGFiMS04MmRjOGEwYTZhMDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDhUMTEzMjMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzhjYzc0MmZkNjI4MzBjMzgwZjdkZmI3NTBkMDIwYjlhYjYwZGJhZjk3Yjk1OTFmNTg4NDUwNzFhNjI2Y2IzMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.pK3z0zK3ZDlkPL0frXeieqiwsiZF8QrFn19HVMu8RF8">

Thank you for your quick reply! I upgraded to the latest `earthkit` and now it seems to work :)
